File: ABBA/solve/cpp/ABBASolver.cpp
#include <fstream>
#include <topcoder>
#include "ABBA.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
    try {
        std::ifstream ifs(argv[1]);
        string initial; tc::read(ifs, initial); tc::next(ifs);
        string target; tc::read(ifs, target);
        ifs.close();

        std::ofstream ofs(argv[2]);
        ABBA solver;
        tc::show(ofs, solver.canObtain(initial, target));
        ofs.close();
    } catch (std::exception &e) {
        std::cerr << e.what() << std::endl;
    }
    return 0;
}

File: ABBA/solve/java/ABBASolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class ABBASolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        String initial = (String) reader.next(String.class);
        reader.next();
        
        String target = (String) reader.next(String.class);
        reader.close();

        ABBA solver = new ABBA();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.canObtain(initial, target));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}
